sdk/eventgrid/azure-messaging-eventgrid/src/test/java/com/azure/messaging/eventgrid/EventGridPublisherImplTests.java

```diff
@@ -0,0 +1,166 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for
+// license information.
+
+package com.azure.messaging.eventgrid;
+
+import com.azure.core.http.HttpHeaders;
+import com.azure.core.http.HttpPipelineBuilder;
+import com.azure.core.http.policy.AddHeadersPolicy;
+import com.azure.core.http.policy.RetryPolicy;
+import com.azure.core.http.rest.Response;
+import com.azure.core.test.TestBase;
+import com.azure.messaging.eventgrid.implementation.EventGridPublisherClientImpl;
+import com.azure.messaging.eventgrid.implementation.EventGridPublisherClientImplBuilder;
+import com.azure.messaging.eventgrid.implementation.models.CloudEvent;
+import com.azure.messaging.eventgrid.implementation.models.EventGridEvent;
+import com.fasterxml.jackson.core.JsonParser;
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.DeserializationContext;
+import com.fasterxml.jackson.databind.JsonDeserializer;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.module.SimpleModule;
+import org.junit.jupiter.api.Test;
+
+import java.io.IOException;
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.time.OffsetDateTime;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.UUID;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+
+
+public class EventGridPublisherImplTests extends TestBase {
+
+    HttpPipelineBuilder builder;
+
+    private static final String EVENTGRID_ENDPOINT = "EG_ENDPOINT";
+
+    private static final String CLOUD_ENDPOINT = "EG_CLOUD_ENDPOINT";
+
+    private static final String CUSTOM_ENDPOINT = "EG_CUSTOM_ENDPOINT";
+
+    private static final String EVENTGRID_KEY = "EG_KEY";
+
+    private static final String CLOUD_KEY = "EG_CLOUD_KEY";
+
+    private static final String CUSTOM_KEY = "EG_CUSTOM_KEY";
+
+    private static final String DUMMY_ENDPOINT = "https://www.dummyEndpoint.com";
+
+    private static final String DUMMY_KEY = "dummyKey";
+
+    @Override
+    protected void beforeTest() {
+        builder = new HttpPipelineBuilder();
+
+        if (interceptorManager.isPlaybackMode()) {
+            builder.httpClient(interceptorManager.getPlaybackClient());
+        } else {
+            builder.policies(interceptorManager.getRecordPolicy(), new RetryPolicy());
+        }
+    }
+
+    @Test
+    public void testPublishEventGridEventsImpl() throws MalformedURLException {
+        EventGridPublisherClientImpl egClient = new EventGridPublisherClientImplBuilder()
+            .pipeline(builder.policies(
+                new AddHeadersPolicy(new HttpHeaders().put("aeg-sas-key", getKey(EVENTGRID_KEY))))
+                .build())
+            .buildClient();
+
+        List<EventGridEvent> events = new ArrayList<>();
```

Comment:
This can be Collections.singletonList() iirc so you don't have to allocate an ArrayList to add a single item.